assign a function variable

The following program `sample.go` triggers a panic:

```go
package main
  
type foo func(b int)

func boo(b int) { println("boo", b) }

func main() {
    var f foo

    f = boo
    f(4)
}
```

Expected result:
```console
$ go run ./sample.go
boo 4
```

Got:
```console
$ yaegi ./sample.go
/home/marc/tmp/fail_f.go:8:6: panic
panic: reflect.Set: value of type *interp.node is not assignable to type func(int) [recovered]
        panic: reflect.Set: value of type *interp.node is not assignable to type func(int)

goroutine 1 [running]:
github.com/containous/yaegi/interp.runCfg.func1(0xc00016e370, 0xc0001e4000)
        /home/marc/go/src/github.com/containous/yaegi/interp/run.go:107 +0x1e9
panic(0xbfeb60, 0xc000170680)
        /usr/lib/go/src/runtime/panic.go:522 +0x1b5
reflect.Value.assignTo(0xc91220, 0xc0001e0a00, 0x16, 0xd1fe64, 0xb, 0xbfc8a0, 0x0, 0xc0001b1cf0, 0x4a4cbd, 0xbfc8a0)
        /usr/lib/go/src/reflect/value.go:2339 +0x437
reflect.Value.Set(0xbfc8a0, 0xc00016c168, 0x193, 0xc91220, 0xc0001e0a00, 0x16)
        /usr/lib/go/src/reflect/value.go:1473 +0xa8
github.com/containous/yaegi/interp.assign.func3(0xc00016e370, 0xc00015e780)
        /home/marc/go/src/github.com/containous/yaegi/interp/run.go:258 +0xb6
github.com/containous/yaegi/interp.runCfg(0xc0001e4000, 0xc00016e370)
        /home/marc/go/src/github.com/containous/yaegi/interp/run.go:112 +0x70
github.com/containous/yaegi/interp.(*Interpreter).run(0xc0001d2000, 0xc0001e1800, 0xc00016e050)
        /home/marc/go/src/github.com/containous/yaegi/interp/run.go:93 +0x129
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc0001d2000, 0xc0001dc000, 0x76, 0x276, 0xc0001dc000, 0x76, 0x0, 0xc304a0)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:294 +0x3f1
main.main()
        /home/marc/go/src/github.com/containous/yaegi/cmd/yaegi/yaegi.go:77 +0x2f8
```

--- a/_test/assign9.go
+++ b/_test/assign9.go
@@ -1,0 +1,15 @@
+package main
+
+type foo func(b int)
+
+func boo(b int) { println("boo", b) }
+
+func main() {
+	var f foo
+
+	f = boo
+	f(4)
+}
+
+// Output:
+// boo 4

--- a/interp/run.go
+++ b/interp/run.go
@@ -1929,7 +1929,7 @@ func reset(n *node) {
 
 	switch l := len(n.child) - 1; l {
 	case 1:
-		typ := n.child[0].typ.TypeOf()
+		typ := n.child[0].typ.frameType()
 		i := n.child[0].findex
 		n.exec = func(f *frame) bltn {
 			f.data[i] = reflect.New(typ).Elem()
@@ -1938,7 +1938,7 @@ func reset(n *node) {
 	case 2:
 		c0, c1 := n.child[0], n.child[1]
 		i0, i1 := c0.findex, c1.findex
-		t0, t1 := c0.typ.TypeOf(), c1.typ.TypeOf()
+		t0, t1 := c0.typ.frameType(), c1.typ.frameType()
 		n.exec = func(f *frame) bltn {
 			f.data[i0] = reflect.New(t0).Elem()
 			f.data[i1] = reflect.New(t1).Elem()
@@ -1949,7 +1949,7 @@ func reset(n *node) {
 		index := make([]int, l)
 		for i, c := range n.child[:l] {
 			index[i] = c.findex
-			types[i] = c.typ.TypeOf()
+			types[i] = c.typ.frameType()
 		}
 		n.exec = func(f *frame) bltn {
 			for i, ind := range index {

--- a/interp/type.go
+++ b/interp/type.go
@@ -752,8 +752,11 @@ func (t *itype) TypeOf() reflect.Type {
 	return t.rtype
 }
 
-func (t *itype) frameType() reflect.Type {
-	var r reflect.Type
+func (t *itype) frameType() (r reflect.Type) {
+	var err error
+	if t, err = t.finalize(); err != nil {
+		panic(err)
+	}
 	switch t.cat {
 	case arrayT:
 		if t.size > 0 {

